pkg/shell/util.tsx

```diff
@@ -91,6 +91,16 @@ export interface ManifestItem {
     keywords: ManifestKeyword[];
 }
 
+export interface ManifestParentSection {
```

Comment:
Hmm... we could make some functions like `manifest_parent_section(m: Manifest): ManifestParentSection` which would ide all the "unsafe" JSON manipulations.

Reply:
The whole Manifest stuff should go to its own file, actually.

Reply:
There is big WIP commit now that implements one approach: validating all manifests as they are received from the bridge.  It's quite a bit of code, and it totally wants to be autogenerated, but I think that's not worth it just yet.  (Also, it's not enough code to justify dragging in some toolkit/framework for this, unless we find a really nice one.)

Reply:
Addendum: the validation is pretty fine-grained: If you forget the label of a docs entry, only that one docs entry is omitted. If you use a string for the weight of a keyword instead of a number, you get an error message but the shell continues with a weight of 0.